src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolder.java

```diff
@@ -705,10 +702,21 @@ public void doConfigSubmit(StaplerRequest req, StaplerResponse rsp) throws IOExc
             if (namingStrategy.isForceExistingJobs()) {
                 namingStrategy.checkName(name);
             }
-            FormApply.success(".").generateResponse(req, rsp, this);
+            FormApply.success(getSuccessfulDestination()).generateResponse(req, rsp, this);
         }
     }
 
+    /**
+     * Where user will be redirected after creating a {@code AbstractFolder}.
```

Comment:
They are separate RFEs. I am just pointing out that

* the current Javadoc only covers one of the two cases where this will be called creation but not reconfiguration)
* if JENKINS-18681 is implemented, for (non-`Computed`) `Folder`s this method will _not_ be called during creation